Sources/PrinceOfVersions/PrinceOfVersions.swift

```diff
@@ -42,57 +29,94 @@ public extension PrinceOfVersions {
 
      After check with server is finished, this method will return all informations about the app versioning.
      It's up to the user to handle that info in a way sutable for the app.
-     For automated check if new optional or mandatory version is available, you can use `checkForUpdates` method.
      If configuration URL could not be provided, you can use `checkForUpdateFromAppStore` for automated check if new version is available.
 
      - parameter URL: URL that containts configuration data.
-     - parameter httpHeaderFields: Optional HTTP header fields.
-     - parameter shouldPinCertificates: Boolean that indicates whether PoV should use security keys from all certificates found in the main bundle. Default value is `false`.
      - parameter callbackQueue: The queue on which the completion handler is dispatched. By default, `main` queue is used.
-     - parameter completion: The completion handler to call when the load request is complete. It returns result that contains UpdatInfo data or PrinceOfVersionsError error
+     - parameter options: Used for additional configuration such as `shouldPinCertificates`, `httpHeaderFields` and `userRequirements`
+     - parameter completion: The completion handler to call when the load request is complete. It returns result that contains UpdateResult data or PoVError error
 
      - returns: Discardable `URLSessionDataTask`
      */
     @discardableResult
-    func loadConfiguration(
+    static func checkForUpdates(
```

Comment:
Red Riding Hood says "What big function you have!" and the wolf replies "The better to see it all!" Red Riding Hood says "What big nested functions you have!" and the wolf replies "The better to confuse you with!"